scripts/cnn_variant_wdl/cram2filtered.wdl

```diff
@@ -0,0 +1,473 @@
+    }
+}
+
+task MergeVCFs {
+    # inputs
+    Array[File] input_vcfs
+    String output_prefix
+
+    File? gatk_override
+
+    # runtime
+    String gatk_docker
+    Int? mem
+    Int? preemptible_attempts
+    Int? disk_space
+    Int? cpu
+    Boolean use_ssd = true
+
+    # Mem is in units of GB but our command and memory runtime values are in MB
+    Int machine_mem = if defined(mem) then mem * 1000 else 3500
+    Int command_mem = machine_mem - 1000
+
+    # using MergeVcfs instead of GatherVcfs so we can create indices
+    # WARNING 2015-10-28 15:01:48 GatherVcfs  Index creation not currently supported when gathering block compressed VCFs.
+    command {
+        set -e
+        export GATK_LOCAL_JAR=${default="/root/gatk.jar" gatk_override}
+        #gatk --java-options "-Xmx${command_mem}m" MergeVcfs \
+        java "-Xmx${command_mem}m" -jar ${gatk_override} MergeVcfs \
+        -I ${sep=' -I ' input_vcfs} -O "${output_prefix}_cnn_scored.vcf.gz"
+    }
+
+    runtime {
+        docker: gatk_docker
+        memory: machine_mem + " MB"
+        disks: "local-disk " + select_first([disk_space, 100]) + if use_ssd then " SSD" else " HDD"
+        preemptible: select_first([preemptible_attempts, 10])
+        cpu: select_first([cpu, 1])
+    }
+
+    output {
+        File merged_vcf = "${output_prefix}_cnn_scored.vcf.gz"
+        File merged_vcf_index = "${output_prefix}_cnn_scored.vcf.gz.tbi"
+    }
+}
+
+
+task SamtoolsMergeBAMs {
+    Array[File] input_bams
+    String output_prefix
+    File picard_jar
+    command {
+        samtools merge ${output_prefix}_bamout.bam ${sep=' ' input_bams} 
+    }
+
+    output {
+        File bamout = "${output_prefix}_bamout.bam"
+        File bamout_index = "${output_prefix}_bamout.bai"
+    }
+
+  runtime {
+    docker: "broadinstitute/genomes-in-the-cloud:2.1.1"
+    memory: "16 GB"
+    disks: "local-disk 400 HDD"
+  }    
+}
+
+
+task MergeBamOuts {
+    # inputs
+    File ref_fasta
+    File ref_fai
+    File ref_dict
+    Array[File]+ bam_outs
+    String output_prefix
+
+    File? gatk_override
+
+    # runtime
+    String gatk_docker
+    Int? mem
+    Int? preemptible_attempts
+    Int? disk_space
+    Int? cpu
+    Boolean use_ssd = true
+
+    # Mem is in units of GB but our command and memory runtime values are in MB
+    Int machine_mem = if defined(mem) then mem * 1000 else 7000
+    Int command_mem = machine_mem - 1000
+
+    command <<<
+        # This command block assumes that there is at least one file in bam_outs.
+        #  Do not call this task if len(bam_outs) == 0
+        set -e
+        export GATK_LOCAL_JAR=${default="/root/gatk.jar" gatk_override}
+        #gatk --java-options "-Xmx${command_mem}m" GatherBamFiles \
+        java "-Xmx${command_mem}m" -jar ${gatk_override} GatherBamFiles \
+            -I ${sep=" -I " bam_outs} -O unsorted.out.bam -R ${ref_fasta}
+
+        # We must sort because adjacent scatters may have overlapping (padded) assembly regions, hence
```

Comment:
The can happen within a scatter actually.  Change adjacent scatters to adjacent variants.